api/nodes/views.py

```diff
@@ -25,7 +26,11 @@ def get_node(self):
         obj = get_object_or_404(Node, self.kwargs[self.node_lookup_url_kwarg])
         # May raise a permission denied
         self.check_object_permissions(self.request, obj)
-        return obj
+
+        if obj.is_deleted:
```

Comment:
It may make sense for this logic to go in `get_object_or_404` (though we would need to update the function name to something like `get_object_or_error`. That way, we can generalize this for other models that have an `is_deleted` attribute.